Update Play to Version 2.6

felixoi opened the pull request:
Fixes #468

felixoi committed:
Updated dependencies

felixoi committed:
Breaking all the things

felixoi committed:
Added 200 more errors

Faithcaio committed:
get stuff loading again

Faithcaio committed:
fix more config warnings/errors

felixoi committed:
Fixed http secret

felixoi committed:
More fixes

phase commented on app/controllers/ApiController.scala:
```diff
@@ -145,7 +147,7 @@ final class ApiController @Inject()(api: OreRestfulApi,
   private def error(key: String, error: String) = Json.obj("errors" -> Map(key -> List(this.messagesApi(error))))
 
-  def deployVersion(version: String, pluginId: String, name: String) = ProjectAction(pluginId) { implicit request =>
+  def deployVersion(version: String, pluginId: String, name: String) = (Action andThen projectAction(pluginId)) { implicit request =>
```
Why was `ProjectAction` removed here?

felixoi replied:
The behavior is not removed. projectAction does exactly the same, we had to change this because of a the action-refactor in 2.6.

phase replied:
But [`OreBaseController` has `ProjectAction`](https://github.com/SpongePowered/Ore/pull/498/files#diff-022b1f4b21acedf492f737eb21692398R89) in it.

phase commented on app/controllers/project/Projects.scala:
```diff
@@ -20,7 +20,7 @@ import ore.project.factory.ProjectFactory
 import ore.user.MembershipDossier._
 import ore.{OreConfig, OreEnv, StatTracker}
-import play.api.cache.CacheApi
+import play.api.cache.{CacheApi, SyncCacheApi}
```
`CacheApi` can be removed because it's not used anywhere in the class.

phase commented on app/util/JavaUtils.scala:
```diff
@@ -0,0 +1,15 @@
+
+object JavaUtils {
+
+  def autoClose[A <: AutoCloseable](closeable: A)(fun: (A) ⇒ Unit)(exHandler: Exception => Unit): Unit = {
```
Is this really needed if we're only using it in one place?

felixoi replied:
I think that's fine. It's a very abstract method we could need if we extend some features.

phase commented on app/views/users/admin/queue.scala.html:
```diff
@@ -176,7 +175,7 @@ <h3><i class="fa fa-thumbs-o-up"></i> @messages("user.queue.none")</h3>
         $(function(){
             var momentNow = moment();
-            var maxDiffrence = @config.ore.getConfig("queue").get.getInt("max-review-time").getOrElse(86400000);
+            var maxDiffrence = @config.ore.get[Configuration]("queue").get[Int]("max-review-time")OrElse(86400000);
```
`OrElse` here looks like a typo

phase commented on conf/application.conf.template:
```diff
@@ -25,19 +25,22 @@ application {
     filters.enabled                   += "filters.MimeKeyedCspFilter"
     filters.csrf.body.bufferSize       =   1000000
+
+    http {
```
Someone needs to remind @progwml6 to update the config once this is merged.

phase reviewed (changes requested):
Looks good, I'm glad we can stay up to date with Play versions.

progwml6 commented:
@felixoi  @Faithcaio this PR has merge conflicts

phase committed:
Merge branch 'master' into feature/play-2.6

This fixes the dependencies to use Slick instead of Raven.

phase commented:
I resolved the conflicts (commit not verified because I'm not at my usual computer).

phase committed:
Remove unused import

Makes the diff for the play-2.6 PR a little cleaner.

phase committed:
Clean up some migration changes

ProjectAction was removed when it didn't need to be.

Some things in sugar.Calls weren't annotated with their return type
which was causing errors (in IntelliJ at least).

There was an error with getting a number from the config.

Signed-off-by: Jadon Fowler <jadonflower@gmail.com>

phase committed:
More Play 2.6 Migrations

Remove all uses of CacheApi and replace it with SyncCacheApi.

Replace Tags in the RequestHeaders for Nonces with Attributes.

Signed-off-by: Jadon Fowler <jadonflower@gmail.com>
